UI/Web/src/app/_services/gestures.service.ts

```diff
@@ -0,0 +1,23 @@
+let swipeCoord = [0, 0];
+let swipeTime = new Date().getTime();
+
+export function swipe(e: TouchEvent, when: string): string | undefined {
```

Comment:
Make when type: 'start' | 'end'

Reply:
Changed parameter to a strong, strict, static type.